torax/core_profile_setters.py

```diff
@@ -604,16 +631,39 @@ def _init_psi_and_current(
   Returns:
     Refined core profiles.
   """
-  # Retrieving psi from the profile conditions.
+  use_vloop_as_boundary_condition = (
+      dynamic_runtime_params_slice.profile_conditions.use_vloop_lcfs_boundary_condition
+  )
+
+  # Case 1: retrieving psi from the profile conditions, using the prescribed profile and Ip_tot
   if dynamic_runtime_params_slice.profile_conditions.psi is not None:
+    # Calculate the dpsi/drho necessary to achieve the given Ip_tot
+    dpsi_drhonorm_edge = _calculate_psi_grad_constraint_from_Ip_tot(
+        dynamic_runtime_params_slice,
+        geo,
+    )
+
+    # Set the BCs to ensure the correct Ip_tot
+    if use_vloop_as_boundary_condition:
```

Comment:
I know this was true already but there is now a lot of repeated/confusing code here

Can we maybe look tomorrow at a way to refactor/filter this a bit

And also apologies in advance that there may be some more merge conflicts here, we have obviously all decided to fix the same bit of code

Reply:
Definitely agree, I'm not particularly happy with how this function has ended up looking. 
I'm currently trying to fix up the unit tests, and then will try and achieve ~zero merge conflicts before tomorrow so that we can focus on refactoring.